src/main/java/com/uber/cadence/internal/sync/POJOActivityTaskHandler.java

```diff
@@ -197,7 +207,45 @@ public Result handle(
         metricsScope.gauge(MetricsType.ACTIVITY_ACTIVE_THREAD_COUNT).update(Thread.activeCount());
       }
     }
-    return activity.execute(activityTask, metricsScope);
+
+    ActivityTaskHandler.Result activityResult;
+    try {
+      Map<String, String> mdcContextMap = MDC.getCopyOfContextMap();
+      List<Future<ActivityTaskHandler.Result>> futures =
+          Executors.newSingleThreadExecutor()
+              .invokeAll(
+                  Collections.singletonList(
+                      () -> {
+                        contextPropagationHandler.refreshContext();
```

Comment:
This is needed for propagating ThreadLocal contexts to the thread that will be executing the task.